OpenRA.Mods.Common/ActorIndex.cs

```diff
@@ -111,22 +112,22 @@ public NamesAndTrait(World world, IReadOnlyCollection<string> names)
 
 			static IEnumerable<Actor> ActorsToIndex(World world, HashSet<string> names)
 			{
-				return world.ActorsHavingTrait<T>().Where(a => names.Contains(a.Info.Name));
+				return world.Actors.Where(a => names.Contains(a.Info.Name) && a.Info.HasTraitInfo<TTraitInfo>());
```

Comment:
isn't the former check constant time while the latter iterative? Maybe switch the checks for perf?

Reply:
Both are hash lookups (`HashSet` for `names.Contains` and `Dictionary` for `Info.HashTraitInfo`) so no major difference.

We generally expect the names check to be more selective - i.e. return false more often, so it makes sense to run that first in hopes of being able to skip the later check.

Reply:
Actors is not a HashSet